fawltydeps/packages.py

```diff
@@ -229,20 +230,31 @@ def determine_package_dir(cls, path: Path) -> Optional[Path]:
         # From there, the returned directory is that site-packages subdir.
         # Note that we must also accept lib/pythonX.Y/site-packages for python
         # versions X.Y that are different from the current Python version.
+        found = False
         if (path / "bin/python").is_file():
             for site_packages in path.glob("lib/python?.*/site-packages"):
                 if site_packages.is_dir():
-                    return site_packages
+                    yield site_packages
+                    found = True
+            if found:
+                return
 
         # Workaround for projects using PEP582:
         if path.name == "__pypackages__":
             for site_packages in path.glob("?.*/lib"):
                 if site_packages.is_dir():
-                    return site_packages
+                    yield site_packages
+                    found = True
+            if found:
+                return
 
         # Given path is not a python environment, but it might be _inside_ one.
         # Try again with parent directory
-        return None if path.parent == path else cls.determine_package_dir(path.parent)
+        if path.parent != path:
+            for package_dir in cls.find_package_dirs(path.parent):
+                with suppress(ValueError):
+                    package_dir.relative_to(path)  # ValueError if not relative
+                    yield package_dir
```

Comment:
Why do we need this mechanism?

`package_dir.relative_to(path)` looks like it was called but its outputs were never used.

Reply:
So to sum up this:
```python
                with suppress(ValueError):
                    package_dir.relative_to(path)  # ValueError if not relative
                    yield package_dir
```
is a hacky trick to:
```python
                if package_dir.relative_to(path)
                    yield package_dir
```
which is available only for Python 3.9+?

If yes, then I get it and indeed looks like the next best thing.